### ISSUE TITLE
Discussion: Different tagcloud approach, nested tag views

### ISSUE THREAD

**Issue by thewilli** [shaarli/Shaarli#959]
Currently, if you click on a tag on the tag cloud, you get forwarded to the link list with a corresponding filter set. What do you think of the following modification to the tag cloud:

If you click on any tag in the tag cloud, you are not redirected away, but stay on the tag cloud. A list of links matching the selected tag is shown below the cloud, which now does not longer show all available tags, but only those available with the new (tag-) filter set. The current search is displayed at the top as a kind of "breadcrumps" which allows to undo the last step(s). With each click on an *available* tag, both, the tag cloud and result list get filtered again.

This way one could easily find relations between tag and have another opportunity to search and browse the notes and links.

**Comment by ArthurHoaro** [comment]
If I understand correctly, you'd like to merge the search bar in the tag cloud and the actual tag search? That could be interesting... ping @Lucas-C who worked on this feature.

**Comment by thewilli** [comment]
> If I understand correctly, you'd like to merge the search bar in the tag cloud and the actual tag search?

Exactly.

**Comment by Lucas-C** [comment]
Hi @thewilli

Thanks for the feature suggestion.

First: do you know / are you using the links on the tag clouds **counters** ?
They do something very similar to what you are suggesting I think.
And that helps a lot with finding relations between tags :)

Now if I got it right, your suggestion would be to render both the links list **and** the tag cloud on the same page.
It's interesting, but I think then the page would become very redundant with the "home" search page. An equivalent solution could be to hide / display the tag cloud in the "home" search page with a button. This way one could correlate the tag cloud with the matching list of links, and the dedicated tag cloud page could be removed.

This is just my "logical" train of thoughts. I honestly do not like this solution very much. It would make the "home" search page to bloated IMHO.
My personal preference would be keep things as they are now, with only one links list in the "home" search page, and one tag cloud on its dedicated page.
But that's mostly a matter of taste I guess.

What do you think ?

**Comment by nodiscc** [comment]
> Currently, if you click on a tag on the tag cloud, you get forwarded to the link list with a corresponding filter set. What do you think of the following modification to the tag cloud:
> If you click on any tag in the tag cloud, you are not redirected away, but stay on the tag cloud. A list of links matching the selected tag is shown below the cloud, which now does not longer show all available tags, but only those available with the new (tag-) filter set

As @Lucas-C pointed out, clicking on the number/counter next to a tag `foo` in the Tag Cloud, will _filter the tag cloud_ and only display other tags that appear in shaares tagged `foo`. (`?do=tagcloud&searchtags=foo`)

![xcfsfs](https://user-images.githubusercontent.com/3950390/30086718-e8b964fe-929c-11e7-8d59-05641dc28803.png)


- [ ] Maybe this is not too well documented (tag cloud usage should probably be added here https://github.com/shaarli/Shaarli/blob/master/doc/md/Browsing-and-searching.md).

@thewilli is this what you're looking for?

> The current search is displayed at the top as a kind of "breadcrumps" which allows to undo the last step

For that part, I think directly editing in the text field is fine. It shows you the current search terms, and can be edited easily.

> [ filter the list based on this tagcloud refining/search ]

Clicking on ` List all links with those tags ` will show you a filtered list corresponding to your tag search.

 - [ ] This could be documented/clarified (center the link, add an arrow or list icon next to it.)


> Now if I got it right, your suggestion would be to render both the links list and the tag cloud on the same page.
> I think then the page would become very redundant with the "home" search page

- [ ] The only way I see this working without cluttering the UI, is to make the tag cloud a foldable panel/menu item on the link list page. 

```
 ★ Shaarli | Tag cloud ▾ | Picture wall | Daily
⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺
  foo 18              bar 19                  baz 30                    ] TAG
        blah 1        bleh 4        wat 5                               ] CLOUD
⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺
    [ search text ] [ filter by tag ] [o]
⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺
2091 results tagged -doc
⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺⎺
```

The tag cloud would fold/unfold when clicking on the arrow next to `Tag cloud`. The advantage is that you could see both the filtered link list and the filtered tag cloud at the same time. However this probably requires rewriting part of how the tag cloud works/load it and hide on every linklist page (performance hit?).

**Comment by Lucas-C** [comment]
> The only way I see this working without cluttering the UI, is to make the tag cloud a foldable panel/menu item on the link list page.

Yes, that's the solution I contemplated, but I still think it would make the links-list page a little bloated, and makes the tagcloud dedicated page useless at the same time

**Comment by thewilli** [comment]
> @thewilli is this what you're looking for?

Yes, kind of, thank you, @nodiscc and @Lucas-C 

> It's interesting, but I think then the page would become very redundant with the "home" search page.  An equivalent solution could be to hide / display the tag cloud in the "home" search page with a button.

Yes, that sounds good.

Maybe I should point out what I finally what to achieve: I am using Shaarli as a kind of small knowledge management system (successively replacing Evernote and DEVONThink for a scientific use-case). The connection of items and tags is therefore very important to me, as finding *related* items is essential - and useful for *regular* links as well.
